sklearn/base.py

```diff
@@ -558,6 +558,35 @@ def fit_transform(self, X, y=None, **fit_params):
             # fit method of arity 2 (supervised transformation)
             return self.fit(X, y, **fit_params).transform(X)
 
+    @property
+    def n_features_out_(self):
```

Comment:
I can't say I like this magic determination. I'd rather it be done by specialised mixins for decomposition and feature selection.

Reply:
Sure, I could do that.
I have to check how much of these are actually in the ``decomposition`` module

Reply:
I'm not sure if I prefer having these be mixins or base classes. It seems unlikely you want to mix those and base classes make the code shorter.